src/vs/editor/common/model/editableTextModel.ts

```diff
@@ -484,9 +484,7 @@ export class EditableTextModel extends TextModelWithDecorations implements Edito
 
 					this.emitModelContentChangedLinesDeletedEvent(spliceStart + 1, spliceStart + spliceCnt);
 					// this.emitModelContentChangedLinesInsertedEvent(startLineNumber + editingLinesCnt + 1, startLineNumber + insertingLinesCnt, newLinesContent.join('\n'));
-				}
-
-				if (editingLinesCnt < insertingLinesCnt) {
+				} else if (editingLinesCnt < insertingLinesCnt) {
```

Comment:
since ```editingLinesCnt``` is the min of the ```insertingLinesCnt``` and ```deletingLinesCnt```, it can only be less than at most one of them, and we want a catch-all else at the end.